src/util/addDecorationsToEditor.ts

```diff
@@ -169,3 +169,45 @@ export function addDecorationsToEditors(
     });
   });
 }
+
+export interface Lexeme {
+  /** The normalised text of the lexeme. */
+  text: string;
+
+  /** The start offset of the lexeme within its containing token */
+  tokenStartOffset: number;
+
+  /** The end offset of the lexeme within its containing token */
+  tokenEndOffset: number;
+}
+
+export function getTokenLexemes(
+  text: string,
+  tokenHatSplittingMode: TokenHatSplittingMode
+): Lexeme[] {
+  const { preserveCase, removeAccents } = tokenHatSplittingMode;
+
+  if (removeAccents) {
+    return matchAll<Lexeme>(text, /\p{L}\p{M}*|\P{L}/gu, (match) => {
+      const matchTextNoAccents = match[0]
+        .normalize("NFD")
+        .replace(/\p{M}/gu, "");
```

Comment:
Tbh, we should probably always split this way and do `NFD` normalisation, even if they want to preserve accents.  And then do the same normalisation when they say "air".  Otherwise they might have weird behaviour where it only sometimes works depending whether the char uses combining mark or single codepoint.  Does that make sense?